cpanspec

```diff
@@ -1746,6 +1798,100 @@ END
+        }
+    }
+
+    %build_requires = %{$meta->{build_requires}} if ($meta->{build_requires});
+    if ($meta->{configure_requires}) {
+        while (my ($key, $value) = each(%{$meta->{configure_requires}})) {
+            if (defined $build_requires{$key}) {
+                next if version->parse($build_requires{$key}) > version->parse($value);
+            }
+            $build_requires{$key} = $value;
+        }
+    }
+    if ($meta->{test_requires}) {
+        while (my ($key, $value) = each(%{$meta->{test_requires}})) {
+            $build_requires{$key} = $value;
+        }
+    }
+
+    %requires   = %{$meta->{requires}}   if ($meta->{requires});
+    %recommends = %{$meta->{recommends}} if ($meta->{recommends});
+    return (\%provides)
+        if (not %build_requires and not %requires and not %recommends);
+    return (\%provides, \%build_requires, \%requires, \%recommends);
+}
+
+sub prereqs_from_metajson {
+    my ($metajson) = @_;
+    my (%provides, %build, %run, %rec);
+    if ($metajson->{provides}) {
+        for my $pkg (keys %{ $metajson->{provides} || {} }) {
+            $provides{ $pkg } = 1;
+        }
+    }
+
+    my $prereqs = $metajson->{prereqs};
+#    delete $prereqs->{develop};
+    return (\%provides) unless keys %$prereqs;
+
+    for my $phase (qw/ build configure test /) {
+        if (my $build = $prereqs->{ $phase }) {
+            my $req = $build->{requires} || {};
+            for my $module (sort keys %$req) {
+                next if exists $build{ $module } and
+                    version->parse($build{ $module }) > version->parse( $req->{ $module });
+                $build{ $module } = $req->{ $module };
+            }
+        }
+    }
+    for my $phase (qw/ runtime /) {
+        if (my $build = $prereqs->{ $phase }) {
+            my $req = $build->{requires} || {};
+            @run{ keys %$req } = values %$req;
+            my $rec = $build->{recommends} || {};
+            @rec{ keys %$rec } = values %$rec;
+        }
+    }
+    return (\%provides, \%build, \%run, \%rec);
+}
+
+sub parse_provides {
+    my ($path, $files) = @_;
+    my %provides;
+    my $scanner = Perl::PrereqScanner->new;
+    foreach my $test (grep /\.(pm|t|PL|pl)/, @$files) {
+        my $doc = PPI::Document->new($path . "/" . $test);
+
+        next unless ($doc);
+
+        # Get the name of the main package
+        my $pkg = $doc->find_first('PPI::Statement::Package');
+        if ($pkg) {
+            $provides{$pkg->namespace} = 1;
+        }
+
+#        my $scan; # = eval { $scanner->scan_ppi_document($doc)->as_string_hash; };
+#        next unless $scan;
+#        my %scanneddeps = %$scan;
+#        foreach my $dep (keys(%scanneddeps)) {
+#            my $ndep = $scanneddeps{$dep};
+#            unless ($build_requires{$dep} && version->parse($build_requires{$dep}) > version->parse($ndep)) {
+#                $possible_build_requires{$dep} = $scanneddeps{$dep};
+#            }
+#        }
```

Comment:
Currently not. Before it was not commented out, but effectively not running because `$scan` was always undef, so I commented out the whole block.
Maybe I can just throw it away completely, because getting the dependencies from parsing is not reliable anyway.

Reply:
At least add a comment to let other people know why the code is disabled.

Reply:
Removed it